nokemobilelibrary/src/main/java/com/noke/nokemobilelibrary/NokeDeviceManagerService.java

```diff
@@ -1588,4 +1606,24 @@ public void useProxy(String address, int port){
         this.port = port;
     }
 
+    private void invalidateConnectionTimer(){
+        currentNoke = null;
+        if(connectionTimer!=null) {
+            connectionTimer.removeCallbacksAndMessages(null);
+        }
+    }
+    private void initializeConnectionTimer(){
+        connectionTimer = new Handler(Looper.getMainLooper());
+        connectionTimer.postDelayed(connectionTimerRunnable(), numberOfSecondsToDetectTheConnectionError*1000);
+    }
+    private Runnable connectionTimerRunnable() {
+        return new Runnable() {
+            @Override
+            public void run() {
+                mGlobalNokeListener.onError(currentNoke, DEVICE_SHUTDOWN_RESULT, "Connection error");
+                disconnectNoke(currentNoke);
+            }
+        };
+    }
+
```

Comment:
I'm open for suggestions on what kind of error you would send to the listener when the time finishes

Reply:
The `DEVICE_SHUTDOWN_RESULT` is a specific result packet from the lock firmware itself, so its usage should be reserved for that.

I recommend creating a new Noke Device Manager Service Error called `ERROR_CONNECTION_TIMEOUT = 317;` and using this.

Reply:
Thanks for the suggestion, making the change